kedro/config/config.py

```diff
@@ -86,18 +86,32 @@ def __init__(
                 This is used in the `conf_paths` property method to construct
                 the configuration paths. Can be overriden by supplying the `env` argument.
         """
-        super().__init__(
-            conf_source=conf_source, env=env, runtime_params=runtime_params
-        )
         self.base_env = base_env
         self.default_run_env = default_run_env
 
+        core_config_patterns = {
+            "catalog": ["catalog*", "catalog*/**", "**/catalog*"],
+            "parameters": ["parameters*", "parameters*/**", "**/parameters*"],
+            "credentials": ["credentials*", "credentials*/**", "**/credentials*"],
+            "logging": ["logging*", "logging*/**", "**/logging*"],
+        }
+        self.config_patterns = core_config_patterns
```

Comment:
Can we include a `config_patterns` argument in the `__init__`, so those patterns are configurable through `settings.py`? Then the code here could look like:

```
self.config_patterns = {
    "catalog": ["catalog*", "catalog*/**", "**/catalog*"],
    "parameters": ["parameters*", "parameters*/**", "**/parameters*"],
    "credentials": ["credentials*", "credentials*/**", "**/credentials*"],
    "logging": ["logging*", "logging*/**", "**/logging*"],
}

self.config_patterns.update(config_patterns or {})
```

Reply:
Yes can do. Originally I intended to have a separation between `core` and `custom` config patterns so it's clear which patterns have been changed by the user:
```
core_patterns = {
            "catalog": ["catalog*", "catalog*/**", "**/catalog*"],
            "parameters": ["parameters*", "parameters*/**", "**/parameters*"],
            "credentials": ["credentials*", "credentials*/**", "**/credentials*"],
            "logging": ["logging*", "logging*/**", "**/logging*"],
        }

if custom_patterns:
    core_patterns.update(custom_patterns)
self.config_patterns = core_patterns
```

But perhaps this way it's also clear that you can actually update the core patterns as well as add your own, e.g. spark config.